From 83e5b47bfc95567cef1cb5bf9b7f50cb2e4c7a55 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Wed, 21 Aug 2024 12:15:56 -0400
Subject: [PATCH] In Python 3.11 and later, use tomllib instead of toml

This removes the dependency on toml for Python versions that have
tomllib in the standard library.
---
 pyproject.toml |  4 ++--
 setup.py       | 13 +++++++++++--
 2 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index ca86fb0..5efc798 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools", "toml"]
+requires = ["setuptools", "toml; python_version < '3.11'"]
 
 [tool]
 [tool.poetry]
@@ -12,7 +12,7 @@ homepage = "https://github.com/gchamon/sysrsync"
 
 [tool.poetry.dependencies]
 python = "^3.6"
-toml = "^0.10.0"
+toml = { version = "^0.10.0", python = "<3.11" }
 
 [tool.poetry.dev-dependencies]
 autopep8 = "*"
diff --git a/setup.py b/setup.py
index f3ad689..5afb00c 100644
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,16 @@
 import setuptools
-import toml
 
-pyproject = toml.load("pyproject.toml")
+try:
+    # Python 3.11 and later
+    import tomllib
+except ImportError:
+    # Python 3.10 and earlier
+    import toml
+
+    pyproject = toml.load("pyproject.toml")
+else:
+    with open("pyproject.toml", "rb") as fh:
+        pyproject = tomllib.load(fh)
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
-- 
2.46.0

